hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieLogRecordReader.java

```diff
@@ -143,22 +145,18 @@ public abstract class BaseHoodieLogRecordReader<T> {
   protected HoodieFileGroupRecordBuffer<T> recordBuffer;
 
   protected BaseHoodieLogRecordReader(HoodieReaderContext readerContext,
-                                      HoodieStorage storage,
+                                      FileGroupReaderState readerState,
```

Comment:
Avoid any changes to `BaseHoodieLogRecordReader` and `HoodieMergedLogRecordReader` and limit the usage of reader state to file group reader and record buffer.